src/theory/builtin/type_enumerator.h

```diff
@@ -75,6 +76,31 @@ class UninterpretedSortEnumerator : public TypeEnumeratorBase<UninterpretedSortE
 
 };/* class UninterpretedSortEnumerator */
 
+
+/** FunctionEnumerator 
+* This enumerates function values, based on the enumerator for the
+* array type corresponding to the given function type.
+*/
+class FunctionEnumerator : public TypeEnumeratorBase<FunctionEnumerator> {
+public:
+  FunctionEnumerator(TypeNode type, TypeEnumeratorProperties * tep = nullptr);
+  /** Get the current term of the enumerator. */
+  Node operator*() override;
+  /** Increment the enumerator. */
+  FunctionEnumerator& operator++() throw() override;
+  /** is the enumerator finished? */
```

Comment:
I am not sure the comments on the overriden functions are informative beyond their contract on TypeEnumeratorInterface. I think they can safely be dropped.

Reply:
I am following the policy that seems to be in place that all functions in h files need comments.  Let me know if this is not the case and if there is a style guideline for this.